paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziCallback.kt

```diff
@@ -122,7 +126,14 @@ internal class PaparazziCallback(
       return LayoutPullParser.createFromFile(File(layoutResource.value))
         .also {
           // For parser of elements included in this parser, publish any aapt declared values
-          aaptDeclaredResources.putAll(it.getAaptDeclaredAttrs())
+          it.getAaptDeclaredAttrs()
+            .also(aaptDeclaredResources::putAll)
+            .onEach { (name) ->
+              val id = --aaptResourceId
```

Comment:
Looking at the Android Studio source they generate aapt resource id's within a [specific range](https://cs.android.com/android-studio/platform/tools/adt/idea/+/mirror-goog-studio-main:android/src/com/android/tools/idea/res/ResourceIdManager.kt;l=100-117) to prevent collisions and also not until `getOrGenerateResourceId` is called since not all ResourceReference's need a resource id. Can we move this to be inserted on demand in our `getOrGenerateResourceId` function?

Reply:
Thanks for the reference and explanation. Replaced it with `DynamicResourceIdManager` in 
961d304429f10c5d45bb1df0810b8fd8840232a0